File: AdminArea/AdminArea.Web/Modules/souqcom/Categorycom/CategorycomForm.cs
using Serenity.ComponentModel;

namespace AdminArea.souqcom.Forms;

[FormScript("souqcom.Categorycom")]
[BasedOnRow(typeof(CategorycomRow), CheckNames = true)]
public class CategorycomForm
{
    public string Name { get; set; }
    public string Image { get; set; }
    public string Description { get; set; }
}

File: AdminArea/AdminArea.Web/Modules/ServerTypes/souqcom/CategorycomForm.ts
import { StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface CategorycomForm {
    Name: StringEditor;
    Image: StringEditor;
    Description: StringEditor;
}

export class CategorycomForm extends PrefixedContext {
    static readonly formKey = 'souqcom.Categorycom';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategorycomForm.init)  {
            CategorycomForm.init = true;

            var w0 = StringEditor;

            initFormType(CategorycomForm, [
                'Name', w0,
                'Image', w0,
                'Description', w0
            ]);
        }
    }
}